.github/workflows/integrationTest.yml

```diff
@@ -370,42 +367,53 @@ jobs:
-        run: >
-          echo run terraform and execute test code &&
-          terraform apply --auto-approve
-          -var="ssh_key=${PRIVATE_KEY}"
-          -var="github_repo=${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}.git"
-          -var="github_sha=${GITHUB_SHA}"
-          -var="install_agent=${{ matrix.arrays.installAgentCommand }}"
-          -var="user=${{ matrix.arrays.username }}"
-          -var="ami=${{ matrix.arrays.ami }}"
-          -var="ca_cert_path=${{ matrix.arrays.caCertPath }}"
-          -var="arc=${{ matrix.arrays.arc }}"
-          -var="binary_name=${{ matrix.arrays.binaryName }}"
-          -var="local_stack_host_name=${{ needs.StartLocalStack.outputs.local_stack_host_name }}"
-          -var="s3_bucket=${S3_INTEGRATION_BUCKET}"
-          -var="vpc_security_group_ids=${VPC_SECURITY_GROUPS_IDS}"
-          -var="key_name=${KEY_NAME}"
-          -var="test_name=cw-integ-test-${{ matrix.arrays.os }}"
-          -var="iam_instance_profile=${IAM_ROLE}"
-          -var="tag=${{ matrix.arrays.tag }}"
-
+        uses: nick-invision/retry@v2
+        with:
+          max_attempts: 3
+          timeout_minutes: 15
+          command: |
+            cd integration/terraform/ec2/linux
+            terraform init
+            if terraform apply --auto-approve \
+              -var="ssh_key=${PRIVATE_KEY}" -var="github_repo=${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}.git" \
+              -var="github_sha=${GITHUB_SHA}" -var="install_agent=${{ matrix.arrays.installAgentCommand }}" \
+              -var="user=${{ matrix.arrays.username }}" \
+              -var="ami=${{ matrix.arrays.ami }}" \
+              -var="ca_cert_path=${{ matrix.arrays.caCertPath }}" \
+              -var="arc=${{ matrix.arrays.arc }}" \
+              -var="binary_name=${{ matrix.arrays.binaryName }}" \
+              -var="local_stack_host_name=${{ needs.StartLocalStack.outputs.local_stack_host_name }}" \
+              -var="s3_bucket=${S3_INTEGRATION_BUCKET}" \
+              -var="vpc_security_group_ids=${VPC_SECURITY_GROUPS_IDS}" \
+              -var="key_name=${KEY_NAME}" \
+              -var="test_name=cw-integ-test-${{ matrix.arrays.os }}" \
+              -var="iam_instance_profile=${IAM_ROLE}" \
+              -var="tag=${{ matrix.arrays.tag }}" ; then terraform destroy -auto-approve
+            else
+              terraform destroy -auto-approve && exit 1
+            fi
+
+      #This is here just in case workflow cancel
       - name: Terraform destroy
-        if: ${{ always() && steps.ec2-linux-integration-test.outputs.cache-hit != 'true' }}
-        run: terraform destroy --auto-approve
+        if: ${{ cancelled() && steps.ec2-linux-integration-test.outputs.cache-hit != 'true' }}
+        uses: nick-invision/retry@v2
+        with:
+          max_attempts: 3
+          timeout_minutes: 8
+          retry_wait_seconds: 20
+          command: cd integration/terraform/ec2/linux && terraform destroy --auto-approve
```

Comment:
Can't we use a working-directory config for the GitHub action to specify this directory instead of calling `cd` to get there? I see that you deleted it from the top level, but AFAIK you can specify working-directory on individual actions

Reply:
It does not work for this retry actions. The retry action always starts at the base dir :/

Reply:
Boo

Reply:
Is there a different GitHub action that does have that? 😄

Reply:
https://github.com/marketplace/actions/retry-action is the other retry action gh supports but I can't seem to get it work with multiple commands :/